### PR TITLE
Various fixes in neurodamus recipes with CoreNEURON

### PR DESCRIPTION
**Pull request by iomaganaris**
- Fixes issue when building neurodamus-core+coreneuron
- Link with coreneuron only when there are dependencies
- Use uname -p like CMake to determine the directory of the installation of special for NEURON and CoreNEURON
More info: https://cmake.org/cmake/help/v3.7/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html ( #848 )
- Fix nmodl dependencies in bison and flex
- Bump deployed flex and bison versions
- Added netpyne package

Second take on: https://github.com/BlueBrain/spack/pull/863

We keep building `libcorenrnmech_<model>.so` because in the future it might be used when running `CoreNEURON` from `NEURON` which only includes only the mod files of the `neurodamus_<model>`

### CODE REVIEW

**Review by jorblancoa — DISMISSED** [review]
LGTM!

File: deploy/packages/python-packages.yaml [hunk 1]
```diff
@@ -139,7 +139,7 @@ packages:
       - python
     specs:
       - py-numpy
-      - py-scipy
+      - py-scipy%gcc
```
Comment (matz-e) [inline deploy/packages/python-packages.yaml]: Uhm… `%gcc` is already included in `target_matrix`

Comment (iomaganaris) [reply deploy/packages/python-packages.yaml]: True true.. I will revert the change

**Comment by iomaganaris** [comment]
retest this please Jenkins

File: var/spack/repos/builtin/packages/py-netpyne/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    depends_on('py-future', type='run')
+    depends_on('py-matplotlib@2.2:', type='run')
+    depends_on('py-matplotlibscalebar', type='run')
+    depends_on('py-numpy', type='run')
+    depends_on('py-pandas', type='run')
+    depends_on('py-scipy', type='run')
+
+    depends_on('python@3.6:')
+    depends_on('neuron+coreneuron+python')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-netpyne/package.py]: could be just run dep

**Review by pramodk — DISMISSED** [review]

**Review by pramodk — APPROVED** [review]

**Review by jorblancoa — APPROVED** [review]

**Pull request merged**